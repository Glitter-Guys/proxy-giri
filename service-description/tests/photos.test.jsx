import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, mount} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Photos from '../client/src/components/Photos';

Enzyme.configure({ adapter: new Adapter() });

var url1 = 'https://images.pexels.com/photos/245239/pexels-photo-245239.jpeg?w=940&h=650&dpr=2&auto=compress&cs=tinysrgb';
var url2 = 'https://images.pexels.com/photos/789818/pexels-photo-789818.jpeg?w=940&h=650&dpr=2&auto=compress&cs=tinysrgb';
var photos = [url1, url2];

it('renders correctly', () => {
  const compPhoto = renderer
    .create(<Photos photos={photos} />)
    .toJSON();
  expect(compPhoto).toMatchSnapshot();
});

it('adds the carousel to the page', () => {
  const wrapper = shallow(<Photos photos={photos} />);
  const div = wrapper.find('.photos-container');
  expect(div).toHaveLength(1);
})

it('adds the correct number of photos to the carousel', () => {
  let wrapper = shallow(<Photos photos={photos} />);
  let div = wrapper.find('.photosCarousel');
  expect(div).toHaveLength(2);

  photos.push(url1);

  wrapper = shallow(<Photos photos={photos} />);
  div = wrapper.find('.photosCarousel');
  expect(div).toHaveLength(3);
})