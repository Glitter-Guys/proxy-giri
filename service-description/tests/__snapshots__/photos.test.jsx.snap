// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="photos-container"
>
  <div
    className={undefined}
  >
    <div
      className="carousel carousel-slider"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <button
        className="control-arrow control-prev control-disabled"
        onClick={[Function]}
        type="button"
      />
      <div
        className="slider-wrapper axis-horizontal"
        style={Object {}}
      >
        <ul
          className="slider animated"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "MozTransform": "translate3d(0%,0,0)",
              "MozTransitionDuration": "350ms",
              "MsTransform": "translate3d(0%,0,0)",
              "MsTransitionDuration": "350ms",
              "OTransform": "translate3d(0%,0,0)",
              "OTransitionDuration": "350ms",
              "WebkitTransform": "translate3d(0%,0,0)",
              "WebkitTransitionDuration": "350ms",
              "msTransform": "translate3d(0%,0,0)",
              "msTransitionDuration": "350ms",
              "transform": "translate3d(0%,0,0)",
              "transitionDuration": "350ms",
            }
          }
        >
          <li
            className="slide selected"
            onClick={[Function]}
          >
            <div
              className="photosCarousel"
              style={
                Object {
                  "backgroundImage": "url(https://images.pexels.com/photos/245239/pexels-photo-245239.jpeg?w=940&h=650&dpr=2&auto=compress&cs=tinysrgb)",
                  "backgroundSize": "cover",
                  "height": "337.5px",
                  "width": "600px",
                }
              }
            />
          </li>
          <li
            className="slide"
            onClick={[Function]}
          >
            <div
              className="photosCarousel"
              style={
                Object {
                  "backgroundImage": "url(https://images.pexels.com/photos/789818/pexels-photo-789818.jpeg?w=940&h=650&dpr=2&auto=compress&cs=tinysrgb)",
                  "backgroundSize": "cover",
                  "height": "337.5px",
                  "width": "600px",
                }
              }
            />
          </li>
        </ul>
      </div>
      <button
        className="control-arrow control-next"
        onClick={[Function]}
        type="button"
      />
      <ul
        className="control-dots"
      >
        <li
          className="dot selected"
          onClick={[Function]}
          value={0}
        />
        <li
          className="dot"
          onClick={[Function]}
          value={1}
        />
      </ul>
    </div>
  </div>
</div>
`;
